kernel/src/memory/mcache.rs

```diff
@@ -50,7 +50,7 @@ sa::const_assert!(core::mem::align_of::<FrameCacheEarly>() <= super::LARGE_PAGE_
 ///
 /// Holds two stacks of pages for O(1) allocation/deallocation.
 /// Implements the `GrowBackend` to hand pages out.
-pub(crate) struct MCache<const BP: usize, const LP: usize> {
+pub struct MCache<const BP: usize, const LP: usize> {
```

Comment:
any reason why this can't be pub(crate) anymore?

Reply:
I believe so. I instantiate a type based on mcache in the rackscale crate. I tried using a separate type that was pub based on mcache, but it still complained about leaking mcache outside the crate if mcache wasn't also pub.

Unless I imported mcache wrong or something in a way that confused rust as to the relation of the mcache/rackscale crates?

Reply:
rackscale should be a module only (not a different crate?) then pub(crate) should make sure it's visible. but if it complains, we can just ignore it and mark it pub only. I found that it's better to make things pub(crate) for dead code elimination checks

Reply:
Ah yes, I double checked why I am getting the error. It is because I was using it in structures I'd marked `pub`. I've just now pushed a commit changing the visibility of the new structures to `pub(crate)`, which allowed me to to revert mcache to `pub(crate)`.